WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -222,13 +220,15 @@ private void launchWithNoteId() {
 
         mViewPager.setCurrentItem(WPMainTabAdapter.TAB_NOTIFS);
 
-        String noteId = getIntent().getStringExtra(NotificationsListFragment.NOTE_ID_EXTRA);
         boolean shouldShowKeyboard = getIntent().getBooleanExtra(NotificationsListFragment.NOTE_INSTANT_REPLY_EXTRA, false);
-
-        if (!TextUtils.isEmpty(noteId)) {
-            NotificationsListFragment.openNote(this, noteId, shouldShowKeyboard, false);
-            GCMIntentService.clearNotificationsMap();
+        if (GCMIntentService.getNotificationsMap().size() == 1) {
```

Comment:
This actually kinda bugged me in a previous PR but I didn't want to seem too picky - and now I guess I feel like being picky :) Is it possible to avoid exposing the notification map outside of `GCMIntentService` (ie: make it private), and then add methods to `GCMIntentService` which supply the desired functionality? It's not a big deal, but I get a minor red flag when I see an object like that exposed when it doesn't need to be.